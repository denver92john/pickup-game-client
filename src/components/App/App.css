.App {
  text-align: center;
}

.App_main {
  margin-top: 50px;
}

.btn {
  margin: 5px 7px;
  padding: 5px 7px;
  border: solid 1px black;
  color: black;
}

nav,
footer {
  padding: 1%;
  background-color: #808080;
}

header {
  padding: 20px 10px;
}

a {
  text-decoration: none;
  color: blue;
}