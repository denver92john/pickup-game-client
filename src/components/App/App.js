import React, {Component} from 'react';
import {Route, Switch} from 'react-router-dom';
import Nav from '../Nav/Nav';
import LandingPage from '../../routes/LandingPage/LandingPage';
import CreatePage from '../../routes/CreatePage/CreatePage';
import DashboardPage from '../../routes/DashboardPage/DashboardPage';
import DiscoverPage from '../../routes/DiscoverPage/DiscoverPage';
import LoginPage from '../../routes/LoginPage/LoginPage';
import SignupPage from '../../routes/SignupPage/SignupPage';
import Footer from '../Footer/Footer';
import STORE from '../../dummy_store';
import './App.css';

class App extends Component {
  render() {
    return (
      <div className="App">
        <Nav />
        <main className="App_main">
          <Switch>
            <Route 
              exact
              path={'/'}
              component={LandingPage}
            />
            <Route 
              path={'/signup'}
              component={SignupPage}
            />
            <Route 
              path={'/login'}
              component={LoginPage}
            />
            <Route 
              path={'/dashboard'}
              store={STORE}
              component={DashboardPage}
            />
            <Route 
              path={'/discover'}
              store={STORE}
              component={DiscoverPage}
            />
            <Route 
              path={'/create'}
              component={CreatePage}
            />
          </Switch>
        </main>
        <Footer />
      </div>
    );
  }
}

export default App;