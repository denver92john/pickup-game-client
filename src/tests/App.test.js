import React from 'react';
import ReactDOM from 'react-dom';
import {BrowserRouter} from 'react-router-dom';
import renderer from 'react-test-renderer';
import App from '../components/App/App';

describe(`App component`, () => {
    it('renders without errors', () => {
        const div = document.createElement('div');
        ReactDOM.render(
            <BrowserRouter>
                <App />
            </BrowserRouter>, 
            div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('renders UI as expected', () => {
        const tree = renderer
            .create(
                <BrowserRouter>
                    <App />
                </BrowserRouter>
            )
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
})