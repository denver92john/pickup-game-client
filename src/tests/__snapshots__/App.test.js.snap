// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component renders UI as expected 1`] = `
<div
  className="App"
>
  <header
    className="App_header"
  >
    <nav
      className="navbar"
      role="navigation"
    >
      <div
        className="logo-div"
      >
        <a
          href="/"
          onClick={[Function]}
        >
          <img
            alt="pickup game logo"
            className="logo"
            src="PUG sports.png"
          />
        </a>
      </div>
      <ul
        className="nav-links"
      >
        <li>
          <a
            className="nav-link"
            href="/signup"
            onClick={[Function]}
          >
            Signup
          </a>
        </li>
        <li>
          <a
            className="nav-link"
            href="/login"
            onClick={[Function]}
          >
            Login
          </a>
        </li>
      </ul>
    </nav>
  </header>
  <main
    className="App_main"
  >
    <div>
      <header
        className="hero "
      >
        <h1
          className="hero-title"
        >
          PickUp Game
        </h1>
        <h3
          className="hero-title"
        >
          Find Your Game
        </h3>
        <p
          className="hero-text"
        >
          PickUp Game, otherwise known as PUG Sports, provides you the ability to find and participate in local pickup games from a variety of different sporting events. Sign up for an event with your friends, meet some new friends, and even try out a new sport.
        </p>
      </header>
      <section
        className="section section-landing-page"
      >
        <header
          className="section-heading"
        >
          <h2>
            Finding a local pickup game
          </h2>
        </header>
        <p
          className="section-sentence"
        >
          Looking for people to play in local pickup games?
        </p>
        <p
          className="section-sentence"
        >
          PickUp Game connects you with others looking to play.
        </p>
      </section>
      <section
        className="section section-landing-page"
      >
        <header
          className="section-heading"
        >
          <h2>
            Creating a local pickup game
          </h2>
        </header>
        <p
          className="section-sentence"
        >
          If you start a game, others will join.
        </p>
      </section>
      <section
        className="section section-landing-page"
      >
        <header
          className="section-heading"
        >
          <h2>
            Demo User Account
          </h2>
        </header>
        <p>
          <b>
            Username: 
          </b>
          JDenver
        </p>
        <p
          className="section-sentence"
        >
          <b>
            Password: 
          </b>
          jdenver
        </p>
        <p
          className="section-sentence"
        >
          To use the demo account, click 
          <a
            href="/login"
            onClick={[Function]}
          >
            'Login'
          </a>
           in the header and enter the above credentials.
        </p>
        <p
          className="section-sentence"
        >
          After logging in, you are taken to the Dashboard view. There will be 2 tabs: My Events and Hosted Events. The My Events tab will display the events you have signed up to participate in. And the Hosted Events tab will show you a list of the events you created and are the host of.
        </p>
        <p
          className="section-sentence"
        >
          The 
          <a
            href="/create"
            onClick={[Function]}
          >
            Create
          </a>
           link in the header will take you to the create an event page. From here you can create an event for others to join.
        </p>
        <p
          className="section-sentence"
        >
          The 
          <a
            href="/discover"
            onClick={[Function]}
          >
            Discover
          </a>
           tab in the header will take you to the discover page. This view will show you a list of all the available events you can click on to preview the event.
        </p>
      </section>
    </div>
  </main>
  <footer
    className="App_footer"
  />
</div>
`;
